src/cocotb_tools/runner.py

```diff
@@ -1506,6 +1506,112 @@ def _test_command(self) -> List[_Command]:
         return cmds
 
 
+class Vcs(Runner):
+    """Implementation of :class:`Runner` for VCS.
+
+    * Does not support the ``pre_cmd`` argument to :meth:`.test`.
+    * Does not support VHDL.
+    """
+
+    supported_gpi_interfaces = {"verilog": ["vpi"]}
+
+    @staticmethod
+    def _simulator_in_path() -> None:
+        if shutil.which("vcs") is None:
+            raise SystemExit("ERROR: vcs executable not found!")
+
+    @staticmethod
+    def _get_include_options(includes: Sequence[PathLike]) -> _Command:
+        return [f"+incdir+{include}" for include in includes]
+
+    @staticmethod
+    def _get_define_options(defines: Mapping[str, object]) -> _Command:
+        return [f"+define+{name}={value}" for name, value in defines.items()]
+
+    def _get_parameter_options(self, parameters: Mapping[str, object]) -> _Command:
+        assert self.hdl_toplevel is not None
+        return [
+            f"-pvalue+{self.hdl_toplevel}.{name}={value}"
+            for name, value in parameters.items()
+        ]
+
+    @property
+    def pli_table_file(self) -> Path:
+        return self.build_dir / "pli.tab"
+
+    @property
+    def pli_table(self) -> Path:
+        return "acc+=rw,wn,cbk:*"
+
+    def _create_pli_file(self) -> None:
+        with open(self.pli_table_file, "w") as f:
+            f.write(self.pli_table)
+
+    @property
+    def sim_file(self) -> Path:
+        return self.build_dir / "simv"
+
+    @property
+    def build_opts(self) -> Path:
+        opts = (
+            ["-full64"]
+            + ["-debug_access+all"]
+            + ["-sverilog"]
+            + ["-LDFLAGS -Wl,--no-as-needed"]
+        )
+
+        if self.verbose:
+            opts += ["-diag all"]
+        else:
+            opts += ["-q"]
+            opts += ["-suppress=VPI-CT-NS"]
+
+        return opts
+
+    def _build_command(self) -> List[_Command]:
+        self._create_pli_file()
+
+        cmds: list[_Command] = []
```

Comment:
This should be `List[_Command]`, subscripting `list` isn't supported until Python 3.8.